lib/widgets/action_sheet.dart

```diff
@@ -372,37 +391,145 @@ class UserTopicUpdateButton extends ActionSheetMenuItemButton {
   }
 }
 
+class ResolveUnresolveButton extends ActionSheetMenuItemButton {
+  ResolveUnresolveButton({
+    super.key,
+    required this.topic,
+    required this.someMessageIdInTopic,
+    required super.pageContext,
+  }) : _actionIsResolve = !topic.isResolved;
+
+  final bool _actionIsResolve;
```

Comment:
nit: order this private field after the public fields that appear in the constructor